### PR TITLE
add settings to control default proposal filters

### PR DESCRIPTION
**Pull request by vyshnavi1605**
## Description
closes: https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/620

## Motivation and Context
Instrument scientists at STFC have to change this to "All proposals" and the status to "All" every time they use the page.

## How Has This Been Tested
Logged on as an Instrument Scientist in the stfc side and tested the view and logged on as an instrument scientist in the ess side and tested the view. Also logged on as a User Officer to check the enabled features. 

image
ESS scientist 

image
STFC scientist 

## Fixes
Fixes: https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/620

## Changes
- Used `PROPOSAL_FILTER` flag to set what filters the instrument scientist should see upon logging in
- Regenerated sdk

## Depends on
https://github.com/UserOfficeProject/user-office-backend/pull/703

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: src/components/proposal/ProposalTableInstrumentScientist.tsx [hunk 1]
```diff
@@ -121,14 +119,24 @@ const ProposalTableInstrumentScientist: React.FC<{
     ...DefaultQueryParams,
     call: NumberParam,
     instrument: NumberParam,
-    proposalStatus: withDefault(NumberParam, 2),
+    proposalStatus: withDefault(
+      NumberParam,
+      featureContext.featuresMap.get(FeatureId.PROPOSAL_FILTER)?.isEnabled
```
Comment (TickleThePanda) [inline src/components/proposal/ProposalTableInstrumentScientist.tsx:124]: Could a setting be better than a feature here? The setting could define `DEFAULT_INST_SCI_STATUS_FILTER` and map properly to the names.

File: src/components/proposal/ProposalTableInstrumentScientist.tsx [hunk 2]
```diff
@@ -121,14 +119,24 @@ const ProposalTableInstrumentScientist: React.FC<{
...
     questionId: StringParam,
     compareOperator: StringParam,
     value: StringParam,
     dataType: StringParam,
     reviewModal: NumberParam,
     modalTab: NumberParam,
-    reviewer: defaultReviewerQueryFilter,
+    reviewer: withDefault(
+      StringParam,
+      featureContext.featuresMap.get(FeatureId.PROPOSAL_FILTER)?.isEnabled
```
Comment (TickleThePanda) [inline src/components/proposal/ProposalTableInstrumentScientist.tsx:136]: Could a separate setting be better than a feature here, too? The setting could define `DEFAULT_INST_SCI_REVIEWER_FILTER` and map to the names?

**Review by martin-trajanovski — APPROVED** [review]

File: src/components/proposal/ProposalTableInstrumentScientist.tsx [hunk 3]
```diff
@@ -111,24 +115,46 @@ const SEPReviewColumns = [
...
 }> = ({ confirm }) => {
   const { user } = useContext(UserContext);
   const featureContext = useContext(FeatureContext);
   const { api } = useDataApiWithFeedback();
+  const { settingsMap } = useContext(SettingsContext);
+  const statusFilterValue = settingsMap.get(
+    SettingsId.DEFAULT_INST_SCI_STATUS_FILTER
+  )?.settingsValue;
+  let statusFilter = proposalStatusFilter.get(statusFilterValue);
+  if (!statusFilter && statusFilter != 0) {
```
[the remark this answers is not in the document]
Comment (vyshnavi1605) [reply src/components/proposal/ProposalTableInstrumentScientist.tsx:136]: `!statusFilter` didn't seem to be allowing `statusFilter = 0` so it was defaulting to 2, so I added that extra check in. But `statusFilter === undefined` alone seems to be working fine, so I'll change that.

File: src/components/proposal/ProposalTableInstrumentScientist.tsx [hunk 4]
```diff
@@ -111,24 +115,46 @@ const SEPReviewColumns = [
   { title: 'SEP', field: 'sepCode', emptyValue: '-', hidden: true },
 ];
 
+const proposalStatusFilter = new Map<Maybe<string> | undefined, Scalars['Int']>(
```
[the remark this answers is not in the document]
Comment (vyshnavi1605) [reply src/components/proposal/ProposalTableInstrumentScientist.tsx:118]: I've tried this, but I get the error `This expression is not callable. Type 'Number' has no call signatures` on the line `let statusFilter = proposalStatusFilter.get(statusFilterValue);` I've tried a few things to fix it but it doesn't seem to work.

[the remark this answers is not in the document]
Comment (TickleThePanda) [reply src/components/proposal/ProposalTableInstrumentScientist.tsx:118]: Ah, sorry - that line would have to be `let statusFilter = proposalStatusFilter[statusFilterValue]` by changing it to a `Record`.

File: src/components/proposal/ProposalTableInstrumentScientist.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (TickleThePanda) [reply src/components/proposal/ProposalTableInstrumentScientist.tsx:136]: I think `statusFilter === undefined || statusFilter === null` is clearer.

File: src/components/review/ReviewerFilter.tsx [hunk 5]
```diff
@@ -15,23 +16,31 @@ const useStyles = makeStyles((theme) => ({
...
-  StringParam,
-  ReviewerFilter.ME
-);
-
 type ReviewerFilterComponentProps = {
   reviewer: string;
   onChange?: (reviewer: ReviewerFilter) => void;
 };
 
+export const reviewFilter = new Map<Maybe<string> | undefined, ReviewerFilter>([
```
Comment (TickleThePanda) [inline src/components/review/ReviewerFilter.tsx:24]: I forgot to say, I think it's worth using a record here, too.

**Review by TickleThePanda — CHANGES_REQUESTED** [review]

File: cypress/integration/appSettings.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ import initialDBData from '../support/initialDBData';
 
 context('App settings tests', () => {
```
Comment (TickleThePanda) [inline cypress/integration/appSettings.ts:3]: Sorry for additional changes. As we've been talking about test coverage, I think these features could need integration tests.

**Review by TickleThePanda — CHANGES_REQUESTED** [review]

**Review by martin-trajanovski — APPROVED** [review]
This seems good to me. 👀 👍

**Review by TickleThePanda — APPROVED** [review]

**Pull request merged**